### PR TITLE
Support ingesting FeedConfigs as entities

### PR DESCRIPTION
**Pull request by deviantintegral**
This adds support for importing FeedConfig objects, making it straight forward for Drupal to react to upstream changes by implementing an event subscriber or an entity hook. This also saves having to refactor the Drush notification commands to support listening for updates outside of a media entity. Finally, the thumbnail download methods are pulled up to the base class so they can be used to download a reasonable thumbnail for feeds.

### CODE REVIEW

File: src/Plugin/media/Source/FeedConfig.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    $feed = $this->getMpxObject($media);
+    $pinned_ids = $feed->getPinnedIds();
+    foreach ($pinned_ids as $video_id) {
+
+      try {
+        $video = $factory->loadByNumericId($video_id)->wait();
+        return $this->downloadThumbnail($video);
+      }
+      catch (ClientException $e) {
+        // Mpx doesn't removed pinned videos from feeds if the video is
```
Comment (juampynr) [inline src/Plugin/media/Source/FeedConfig.php]: Shouldn't this be `doesn't remove`?

**Comment by juampynr** [comment]
Static review looks fine to me. Who should review this, @deviantintegral ?

**Comment by deviantintegral** [comment]
@juampynr I asked @m4olivei to comment on any PRs he had feedback on, as these have mostly been used in production for a few months now.

File: src/Plugin/media/Source/FeedConfig.php (hunk 1 above)
Comment (deviantintegral) [reply src/Plugin/media/Source/FeedConfig.php]: 🤦‍♂

**Pull request merged**